source/jormungandr/jormungandr/parking_space_availability/car/common_car_park_provider.py

```diff
@@ -0,0 +1,103 @@
+# encoding: utf-8
+# Copyright (c) 2001-2018, Canal TP and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Canal TP (www.canaltp.fr).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# IRC #navitia on freenode
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+
+from __future__ import absolute_import, print_function, unicode_literals, division
+
+import logging
+import pybreaker
+import requests as requests
+
+from jormungandr import cache, app, utils, new_relic
+from jormungandr.parking_space_availability import AbstractParkingPlacesProvider
+from abc import abstractmethod
+
+
+class CommonCarParkProvider(AbstractParkingPlacesProvider):
+
+    def __init__(self, operators, dataset, timeout):
+        self.operators = [o.lower() for o in operators]
+        self.timeout = timeout
+        self.dataset = dataset
+        self.breaker = pybreaker.CircuitBreaker(fail_max=self.fail_max, reset_timeout=self.reset_timeout)
+        self.log = logging.LoggerAdapter(logging.getLogger(__name__), extra={'dataset': self.dataset})
+
+    @abstractmethod
+    def _get_information(self, poi):
+        pass
+
+    def get_informations(self, poi):
+        return self._get_information(poi)
+
+    def support_poi(self, poi):
+        properties = poi.get('properties', {})
+        return properties.get('operator', '').lower() in self.operators
+
+    @cache.memoize(app.config['CACHE_CONFIGURATION'].get('TIMEOUT_CAR_PARK', 30))
```

Comment:
minor: i think it may be useful if we can set the timeout for different proxies...

Reply:
As discussed, this is a default value and it is better to keep it common to every car park providers